yellowbrick/features/base.py

```diff
@@ -309,3 +312,130 @@ def _determine_target_color_type(self, y):
                 "could not determine target color type "
                 "from target='{}' to '{}'"
             ).format(self.target_type, self._target_color_type))
+
+
+##########################################################################
+## Projection Visualizers
+##########################################################################
+
+class ProjectionVisualizer(DataVisualizer):
+    
+    def __init__(self, transformer, ax=None, features=None, classes=None, color=None,
+             colormap=None, target_type="auto", proj_dim=2, alpha=0.75, **kwargs):
+        super(ProjectionVisualizer, self).__init__(ax=ax, features=features, 
+                                                     classes=classes, color=color,
+                                                     colormap=colormap, 
+                                                     target_type=target_type, **kwargs)
+        self.transformer=transformer
```

Comment:
Instead of having base classes pass a transformer to `super` (which is a decent idea), I'd prefer the projection visualizer to assume that it is a transformer (e.g. implements the `fit()` and `transform()` methods. This will allow different subclasses to instantiate their transformers whenever they see fit and not just in `__init__` (e.g. they could do it in `fit()`. Manifold and PCA create their transformers very differently so I want to make sure that we can capture all possible use cases.

Reply:
I completely understand your point related to Manifold! Initially, I had the same functionality as you mention here and wanted to discuss it.  
I think that having this parameter allows `ProjectionVisualizer` to be able to project any transformed data in 2D or 3D. By using this, It will act as a different Visualizer that projects when the transformer is given. Although here we can use something like `Visualizer.transformer=PCA`, having a parameter will seem more intuitive. What do you think about this? Let me know your thoughts. I know that this may cause problems with Manifold!

Reply:
Having the transformer as an attribute assumes that the Visualizer _wraps_ a transformer, not that it _is a_ transformer; it's a subtle difference I know. Basically what I'm saying here is that the `ProjectionVisualizer` should just assume that the `transform()` method will correctly project the original data into either two or three dimensions as needed. In fact, it is likely that the `ProjectionVisualizer` does not need `fit`, `transform` or `fit_transform` unless there is a special reason to do so that I don't see quite yet.